lib/internal/util.js

```diff
@@ -187,3 +191,61 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+const kCustomPromisifiedSymbol = Symbol('util.promisify.custom');
+const kCustomPromisifyArgsSymbol = Symbol('customPromisifyArgs');
+
+function promisify(orig) {
+  if (typeof orig !== 'function') {
+    throw new TypeError('The first argument to promisify() must be a function');
+  }
+
+  if (orig[kCustomPromisifiedSymbol]) {
+    return orig[kCustomPromisifiedSymbol];
```

Comment:
What happens in case of the following:

```js
function a(cb) { /* ... */ }
a[util.promisify.custom] = function promisified() {};

const promisified1 = util.promisify(a);
  // promisified1 === a[util.promisify.custom]
const promisified2 = util.promisify(promisified1);
  // promisified2 !== a[util.promisify.custom]
```

Seems like a good idea to define `kCustomPromisifiedSymbol` on the returned function even when there is an existing `orig[kCustomPromisifiedSymbol]`.